src/prog_algs/state_estimators/particle_filter.py

```diff
@@ -52,45 +55,71 @@ def __init__(self, model, x0, measurement_eqn = None, **kwargs):
         # State-estimator specific logic
         if isinstance(self.parameters['n'], Number):
             self.parameters['n'] = {key : self.parameters['n'] for key in self.__measure(x0).keys()}
+        # todo(CT): Check fields on n
+        
+        # Build array inplace
+        x = array(list(x0.values()))
 
-        if isinstance(self.parameters['x0_uncertainty'], Number):
-            # Build array inplace
-            x = array(list(x0.values()))
+        if isinstance(self.parameters['x0_uncertainty'], dict):
```

Comment:
> There's some strange behavior I'm investigating when using the updated PF.
> 
> Is the basic example, using the PF after only one iteration of the estimate function all the samples are identical. I've confirmed looking at the underlying data that every sample is identical. @MikeAndSpencer Do you see this behavior as well?
> 
> <img alt="Screen Shot 2021-10-28 at 4 05 13 PM" width="972" src="https://user-images.githubusercontent.com/8226781/139347901-b138c32d-3aaf-40cb-a2d5-0798424f960e.png">

Good question. This doesn't happen when we I run the particle filter for the electrochemical battery in the particle_filter_battery_example.py script. This is because I set the measurement noise to defined values. The issue you are running into for the base example is that the measurement noise is 0 (because that is the default value given in prog_models). When the measurement noise is 0 the log weights go out to -inf when the particle isn't on the exact mean value. I was able to fix the problem in the basic example by just adding default parameters for the measurement noise in the battery model like so
    # Measurement noise
    R_vars = {
        't': 2, 
        'v': 0.02
    }
    batt = Battery(measurement_noise = R_vars)
I'll include that in a commit.
Spencer